config/opal_config_asm.m4

```diff
@@ -1219,7 +1219,7 @@ int main(int argc, char* argv[])
         [Whether to use builtin atomics])
     AC_SUBST([OPAL_ASSEMBLY_BUILTIN])
 
-    OPAL_SUMMARY_ADD([[Atomics]],[[OMPI]],[],[$opal_cv_asm_builtin])
+    OPAL_SUMMARY_ADD([[Atomics]],[[Built-in flavor]],[],[$opal_cv_asm_builtin])
```

Comment:
I don't think Atomics should have their own top-level category (perhaps "Open MPI Configuration" or "Miscellaneous" would work?).  But we should also add this summary all the time, rather than just when using built-in atomics.

Reply:
I'm reading `opal_config_asm.m4`, and I'm having a helluva time trying to understand it.  ☹️ 

I'm not entirely sure what we should emit if we shouldn't emit `$opal_cv_asm_builtin`.  Can you suggest something else?  Or should we just emit multiple things?  Such as:

```
OPAL_SUMMARY_ADD([[Miscellaneous]],[[Assembly arch]],[],[$opal_cv_asm_arch])
OPAL_SUMMARY_ADD([[Miscellaneous]],[[Built-in atomics flavor]],[],[$opal_cv_asm_builtin])
```

...?

Reply:
Yeah, it's had many people hack on it; I'm not even sure I can completely follow it now.

For now, let's just change this one line to `OPAL_SUMMARY_ADD([[Miscellaneous]],[[Atomics]],[],[$opal_cv_asm_builtin])` and open an issue to make the rest better.

Reply:
@bwbarrett Done!